src/coap_io.c

```diff
@@ -40,6 +40,11 @@
 # include <unistd.h>
 #endif
 #include <errno.h>
+#if defined(HAVE_SYS_EPOLL_H) && defined(HAVE_SYS_TIMERFD_H)
+#include <sys/epoll.h>
+#include <sys/timerfd.h>
+#define HAVE_EPOLL_SUPPORT
```

Comment:
This definition should probably better go into configure.ac because here (in coap_io.c) the epoll-related code depends on both, HAVE_SYS_EPOLL_H and HAVE_SYS_TIMERFD_H, whereas in other files (such as coap_openssl.c, e.g.), it depends on HAVE_SYS_EPOLL_H only.

Reply:
I have got configure.ac to create COAP_EPOLL_SUPPORT if the 2 include files are there and now everything is #ifdef'd using COAP_EPOLL_SUPPORT.